msal/src/main/java/com/microsoft/identity/client/internal/CommandParametersAdapter.java

```diff
@@ -486,4 +492,28 @@ public static GenerateShrCommandParameters createGenerateShrCommandParameters(
                 .popParameters(popParameters)
                 .build();
     }
+
+    /**
+     * Adds additional query string parameters to original list.
+     * @param queryStringParameters extra query string parameters inputted.
+     * @param configuration configuration created from MSAL config file.
+     * @return combined list of query string parameters.
+     */
+    @Nullable
+    private static List<Map.Entry<String, String>> getExtraQueryStringParameters(
+            @Nullable final List<Map.Entry<String, String>> queryStringParameters,
+            @NonNull final PublicClientApplicationConfiguration configuration) {
+        if (configuration.isWebauthnCapable()) {
+            final Map.Entry<String, String> webauthnExtraParameter = new AbstractMap.SimpleEntry<>(
+                    FidoConstants.WEBAUTHN_QUERY_PARAMETER_FIELD,
+                    FidoConstants.WEBAUTHN_QUERY_PARAMETER_VALUE);
+            if (queryStringParameters == null) {
+                return Collections.singletonList(webauthnExtraParameter);
+            }
+            if (!queryStringParameters.contains(webauthnExtraParameter)) {
+                queryStringParameters.add(webauthnExtraParameter);
```

Comment:
>queryStringParameters [](http://example.com/codeflow?start=16&length=21)

If the input list is singletonList() then .add() would fail. Safer to create a new list here.

Reply:
i think it should be fine with the null check + early return above.

Reply:
Hmm just checked AcquireTokenParameters, and looks like mExtraQueryStringParameters is a List, meaning a developer could pass a singletonList in with an entry not related to WebAuthn... and that would lead to an exception thrown here.
Just to be safe, I'll create a new ArrayList here, and I'll also return an ArrayList instead of a SingletonList in the null check in case there are any `add` calls applied to it in the future.

Reply:
See this commit: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1895/commits/fa3f57091d9cca77998282688006513205068b5a Added some unit tests as well.